Adds subplots example in Maps tutorial

navidcy opened the pull request:
This PR adds a  subplots example in the maps tutorial.

navidcy commented:
@DhruvBhagtani could you review this?

@aidanheerdegen, could we add @DhruvBhagtani in the COSIMA org so that he appears in the list of available reviewers to request review from?

navidcy commented:
@aidanheerdegen or @angus-g, why the Sphinx builds fail?

dhruvbhagtani commented:
@navidcy I have made some changes, but I can't commit them to this repository. Probably because I am not a member?

navidcy commented:
Make suggestions then. 
N

> On 14 Dec 2021, at 12:23, Dhruv Bhagtani ***@***.***> wrote:
> 
> ﻿
> @navidcy I have made some changes, but I can't commit them to this repository. Probably because I am not a member?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

angus-g commented:
> ... @angus-g, why the Sphinx builds fail?

looks like GitHub updated the default Python version in the builds to 3.10, which isn't quite compatible with some package we use in our Sphinx build. I've fixed this in master, but you'll probably have to rebase to get that to apply to this PR

navidcy commented:
@AndyHoggANU, you can view the file at
https://nbviewer.org/github/COSIMA/cosima-recipes/blob/bb3bd05a13322d3c0b10f6430ff8dd4be19a4578/Tutorials/Making_Maps_with_Cartopy.ipynb

AndyHoggANU reviewed (commented):
Looks pretty good. Is there an error with the a) b) subplot labelling?

navidcy commented:
There was, I fixed it!

navidcy commented:
![Uploading Screen Shot 2021-12-14 at 3.33.17 pm.png…]()

navidcy commented:
Shall I merge? @dhruvbhagtani?

dhruvbhagtani commented:
This isn't a plotting issue, but wouldn't it be more useful to just set colorer to true? I had corrected the labelling thing but you've done it in the latest commit, so that's great!

navidcy commented:
> This isn't a plotting issue, but wouldn't it be more useful to just set colorer to true? I had corrected the labelling thing but you've done it in the latest commit, so that's great! These were the only two things I changed in my commit, so I don't have any other complaints - it is a really useful example!

Well, if you mean just remove the `add_colorbar=False` then the result looks like this:

<img width="730" alt="Screen Shot 2021-12-14 at 6 11 15 pm" src="https://user-images.githubusercontent.com/7112768/145950194-b3649f1a-4739-4b45-936a-d5c43ffa0148.png">

One must add custom axes for the colorbars, as similarly demonstrated in other parts of the tutorial.

So, I'd say that the all the ingredients are there for people to mix and match from the examples to create the figure of their liking. The non-trivial part was, probably, the `subplot_kw={'projection': ccrs.Robinson()}`.
